### PR TITLE
Add OpenAPI spec with generated client

### PR DESCRIPTION
**Pull request by KamalikaSene**
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

**Comment by ayeshLK** [comment]
`ballerina/openapi.json` should be added into `docs/spec` folder.

**Comment by ayeshLK** [comment]
@KamalikaSene we need to incorporate licensing header into the generated code. If you run following command from the repository root, it would generated client code with licensing header.

```sh
bal openapi -i docs/spec/openapi.yml --mode client --license docs/license.txt -o ballerina
```

**Comment by ayeshLK** [comment]
@KamalikaSene it seems like when generating the client the authentication related configs are not properly set [1] due to a bug in OpenAPI tool. Can you create an issue for the is ballerina-library repo [2] @lnash94 FYI

[1] - https://github.com/ballerina-platform/module-ballerinax-discord/blob/bb9dae023216816ce3f1c7b09e03cd548f7fa1e8/ballerina/client.bal#L42
[2] - https://github.com/ballerina-platform/ballerina-library

**Comment by ayeshLK** [comment]
Lets remove `ballerina/openapi.json` from the PR.

File: ballerina/openapi.json
Comment (NipunaRanasinghe) [inline ballerina/openapi.json:1]: `openapi.json` file should be placed under the `docs/spec` directory. As we already have a copy of this in there, we can remove this.

**Comment by NipunaRanasinghe** [comment]
https://github.com/ballerina-platform/module-ballerinax-slack/pull/151#issuecomment-2188037326

**Comment by NipunaRanasinghe** [comment]
we also need to add a `sanitations.md` file including all the sanitations we did on the original spec and, the exact command we used to generate the client using the openAPI tool. (A example PR with the expected file structures can be found in https://github.com/ballerina-platform/module-ballerinax-asana/pull/2/files)

**Pull request closed without merging**